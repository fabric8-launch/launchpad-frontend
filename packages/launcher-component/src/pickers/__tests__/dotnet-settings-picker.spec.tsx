import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';

import { FormPanel } from '../../core/form-panel/form-panel';
import { DotNetSettingsPicker } from '../dotnet-settings-picker';

afterEach(cleanup);

describe('<DotNetSettingsPicker />', () => {
  it('renders the DotNetSettingsPicker correctly', () => {
    const comp = render(<DotNetSettingsPicker.Element value={{namespace: '', version: ''}} onChange={() => {}}/>);
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it('show error for invalid data', () => {
    const handleSave = jest.fn();
    const comp = render(
      <FormPanel
        initialValue={{}}
        validator={DotNetSettingsPicker.checkCompletion}
        onSave={handleSave}
        onCancel={() => {}}
      >
        {(inputProps) => (<DotNetSettingsPicker.Element {...inputProps}/>)}
      </FormPanel>
    );

    const nameField = comp.getByLabelText('.Net namespace');
    fireEvent.change(nameField, { target: { value: 'invalid name' } });
    const versionField = comp.getByLabelText('.Net version');
    fireEvent.change(versionField, { target: { value: '1' } });
    expect(comp.asFragment()).toMatchSnapshot();
  });
});
