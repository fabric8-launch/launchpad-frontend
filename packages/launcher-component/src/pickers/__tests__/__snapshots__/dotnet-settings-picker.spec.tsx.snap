// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DotNetSettingsPicker /> renders the DotNetSettingsPicker correctly 1`] = `
<DocumentFragment>
  <form
    class="pf-c-form"
    novalidate=""
  >
    <div
      class="pf-c-form__group"
    >
      <label
        class="pf-c-form__label"
        for="name"
      >
        <span
          class="pf-c-form__label-text"
        >
          Namespace
        </span>
      </label>
      <input
        aria-invalid="false"
        aria-label=".Net namespace"
        class="pf-c-form-control"
        id="name"
        name="name"
        pattern="^[a-zA-Z_].*$"
        required=""
        type="text"
        value=""
      />
    </div>
    <div
      class="pf-c-form__group"
    >
      <label
        class="pf-c-form__label"
        for="version"
      >
        <span
          class="pf-c-form__label-text"
        >
          Version
        </span>
      </label>
      <input
        aria-invalid="false"
        aria-label=".Net version"
        class="pf-c-form-control"
        id="version"
        name="version"
        required=""
        type="text"
        value=""
      />
    </div>
  </form>
</DocumentFragment>
`;

exports[`<DotNetSettingsPicker /> show error for invalid data 1`] = `
<DocumentFragment>
  <div
    class="form-panel"
    style="padding: 20px;"
  >
    <form
      class="pf-c-form"
      novalidate=""
    >
      <div
        class="pf-c-form__group"
      >
        <label
          class="pf-c-form__label"
          for="name"
        >
          <span
            class="pf-c-form__label-text"
          >
            Namespace
          </span>
        </label>
        <input
          aria-invalid="false"
          aria-label=".Net namespace"
          class="pf-c-form-control"
          id="name"
          name="name"
          pattern="^[a-zA-Z_].*$"
          required=""
          type="text"
          value="invalid name"
        />
      </div>
      <div
        class="pf-c-form__group"
      >
        <label
          class="pf-c-form__label"
          for="version"
        >
          <span
            class="pf-c-form__label-text"
          >
            Version
          </span>
        </label>
        <input
          aria-invalid="true"
          aria-label=".Net version"
          class="pf-c-form-control"
          id="version"
          name="version"
          required=""
          type="text"
          value="1"
        />
        <div
          aria-live="polite"
          class="pf-c-form__helper-text pf-m-error"
          id="version-helper"
        >
          Please provide a valid version number
        </div>
      </div>
    </form>
    <hr
      class="separator"
    />
    <div
      class="pf-l-toolbar"
    >
      <div
        class="pf-l-toolbar__group"
      >
        <button
          aria-label="Save form"
          class="pf-c-button pf-m-primary pf-m-disabled"
          disabled=""
          type="button"
        >
          Save
        </button>
      </div>
      <div
        class="pf-l-toolbar__group"
      >
        <button
          aria-label="Cancel form"
          class="pf-c-button pf-m-secondary"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
